app/controllers/cart_sessions_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CartSessionsController < ApplicationController
+  before_action :check_quantity, only: :create
+
+  def create
+    sp_create
+    load_products
+    flash.now[:success] = t "carts.create.success"
+  end
+
+  private
+
+  def cart_session_params
+    params.require(:order_product).permit(:product_id, :quantity)
+  end
+
+  def check_quantity
+    if @product = Product.find_by(id: params[:product_id])
+      sp_check_quantity
+    else
+      flash[:danger] = t "carts.create.not_success"
+      redirect_to root_path
+    end
+  end
+
+  def sp_create
+    if !sp_check_id
+      current_cart[params[:product_id].to_s] = params[:quantity].to_f
+    else
+      current_cart[params[:product_id].to_s] += params[:quantity].to_f
+    end
+  end
+
+  def sp_check_quantity
+    check_sum
+    return unless params[:quantity].to_f > @product.quantity.to_f || @db_check
+
+    flash[:danger] = t "carts.create.not_success"
+    redirect_to root_path
+  end
+
+  def check_sum
+    sum = current_cart[params[:product_id].to_s].to_f + params[:quantity].to_f
+    @db_check = @product.quantity < sum
```

Comment:
```suggestion
   @product.quantity < sum
```
cần j tạo biến `@db_check` em, ở line 35 chỉ cần gọi method `check_sum` là được mà

ngoài ra, tạo tên biến đọc chả hiểu
DB check là cái j?  đặt tên biến đọc vô cho nó hiểu hiểu và liên quan tí cậu, phải kéo lên kéo xun đọc hại não nhau

Reply:
dạ em sẽ rút kinh nghiệm